### PR TITLE
Remove cython from install requires

### PR DESCRIPTION
**Pull request by trxcllnt**
Cython is a build-only dependency, so this PR removes Cython from the `install_requires` list of the hdbscan package.

[setup.py reads requirements.txt](https://github.com/scikit-learn-contrib/hdbscan/blob/master/setup.py#L78) to populate its runtime requirements, and [requirements.txt incldues cython](https://github.com/scikit-learn-contrib/hdbscan/blob/master/requirements.txt). So removing it from `requirements.txt` should be sufficient to fix this.

I can make a separate `requirements-build.txt` if the maintainers prefer using a requirements file for build dependencies.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,4 @@
-cython>=0.27,<3
-numpy>=1.20,<2
+numpy>=1.20,<3
```
Comment (jakirkham) [inline requirements.txt:1]: Based on the discussion in issue ( https://github.com/scikit-learn-contrib/hdbscan/issues/642 ) and looking at commit ( https://github.com/scikit-learn-contrib/hdbscan/commit/142c6ac125b72c8433b7915c1816b5fa0e189122 ), maybe it makes sense to keep the NumPy upper bound at 2?

```suggestion
numpy>=1.20,<2
```

**Review by jakirkham — COMMENTED** [review]
Thanks Paul! 🙏

Had a question below

File: requirements.txt (hunk 1 above)
Comment (jakirkham) [reply requirements.txt:1]: Or perhaps I'm mistaken and this works ( https://github.com/scikit-learn-contrib/hdbscan/pull/644 )?

Comment (trxcllnt) [reply requirements.txt:1]: Yeah didn't https://github.com/scikit-learn-contrib/hdbscan/pull/644 fix this?

**Comment by jakirkham** [comment]
Should we change this to use Cython 3+ as well?

https://github.com/scikit-learn-contrib/hdbscan/blob/c201b2e75d42fd0fb7f3aed6b0992afa3c5501c7/pyproject.toml#L5

So...

```diff
-cython<4
+cython>=3,<4
```

**Comment by lmcinnes** [comment]
Thank you!

**Comment by jakirkham** [comment]
Thank you both! 🙏

**Pull request merged**